test/cppsuite/tests/cursor_bound_01.cpp

```diff
@@ -0,0 +1,695 @@
+            testutil_assert(!custom_lexicographical_compare(key, search_key, true));
+        }
+    }
+
+    /*
+     * Validate that if the search key is outside the bounded range, that the range cursor has
+     * returned a record that is either the first or last entry within the range. Do this through
+     * checking if the position of the search key is greater than the range or smaller than the
+     * range. Further perform a next or prev call on the normal cursor and we expect that the key is
+     * outside of the range.
+     */
+    void
+    validate_successful_search_near_outside_range(scoped_cursor &normal_cursor,
+      const bound &lower_bound, const bound &upper_bound, bool larger_search_key)
+    {
+        int ret = larger_search_key ? normal_cursor->next(normal_cursor.get()) :
+                                      normal_cursor->prev(normal_cursor.get());
+        auto lower_key = lower_bound.get_key();
+        auto upper_key = upper_bound.get_key();
+        if (ret == WT_NOTFOUND)
+            return;
+        testutil_assert(ret == 0);
+
+        const char *key;
+        testutil_check(normal_cursor->get_key(normal_cursor.get(), &key));
+        /*
+         * Assert that the next() or prev() call has placed the normal cursor outside of the bounded
+         * range.
+         */
+        auto above_lower_key = lower_key.empty() ||
+          custom_lexicographical_compare(lower_key, key, lower_bound.get_inclusive());
+        auto below_upper_key = upper_key.empty() ||
+          custom_lexicographical_compare(key, upper_key, upper_bound.get_inclusive());
+        testutil_assert(!(above_lower_key && below_upper_key));
+    }
+
+    /*
+     * Validate that the normal cursor is positioned at a key that is outside of the bounded range,
+     * and that no visible keys exist in the bounded range.
+     */
+    void
+    validate_search_near_not_found(
+      scoped_cursor &normal_cursor, const bound &lower_bound, const bound &upper_bound)
+    {
+        int ret, exact;
+        auto lower_key = lower_bound.get_key();
+        auto upper_key = upper_bound.get_key();
+        logger::log_msg(LOG_TRACE,
+          "bounded search_near found WT_NOTFOUND on lower bound: " + lower_key + " upper bound: " +
+            upper_key + " traversing range to validate that there are no keys within range.");
+        if (!lower_key.empty()) {
+            normal_cursor->set_key(normal_cursor.get(), lower_key.c_str());
+            ret = normal_cursor->search_near(normal_cursor.get(), &exact);
+        } else
+            ret = normal_cursor->next(normal_cursor.get());
+
+        testutil_assert(ret == 0 || ret == WT_NOTFOUND);
+
+        if (exact < 0)
+            ret = normal_cursor->next(normal_cursor.get());
+
+        /*
+         * Validate that there are no keys in the bounded range that the range cursor could have
+         * returned.
+         */
+        const char *key;
+        while (true) {
```

Comment:
Should it be `while(ret != WT_NOTFOUND && below_upper_key)`? You will need to define `below_upper_key= true` just before the while loop.

Reply:
Personally I rather just check if `ret != WT_NOTFOUND`, as it might be confusing to have both coniditons here.

Reply:
Sounds good!